### PR TITLE
FLEXURE FOR IFU

### PR DESCRIPTION
**Pull request by debora-pe**
My recent changes to flexure correction were not working for IFU. This PR fixes this. I'm starting to run the tests.

### CODE REVIEW

**Comment by debora-pe** [comment]
Test pass. The one failure was my mistake to run the test using the nirspec Dev Suite branch. It can be ignored.
```
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  249 passed, 66 warnings in 344.02s (0:05:44) ---
--- PYTEST UNIT TESTS PASSED  133 passed, 170 warnings in 1618.43s (0:26:58) ---
--- PYTEST VET TESTS PASSED  61 passed, 97 warnings in 5172.25s (1:26:12) ---
--- PYPEIT DEVELOPMENT SUITE FAILED 1/238 TESTS  ---
Failed tests:
    keck_nirspec_high/Jband pypeit
Skipped tests:
Testing Started at 2024-03-11T09:09:51.866935
Testing Completed at 2024-03-11T22:00:00.073291
Total Time: 12:50:08.206356
```

File: pypeit/find_objects.py [hunk 1]
```diff
@@ -1188,8 +1188,8 @@ def calculate_flexure(self, global_sky):
         sl_ref = self.par['calibrations']['flatfield']['slit_illum_ref_idx']
         box_rad = self.par['reduce']['extraction']['boxcar_radius']
         trace_spat = 0.5 * (self.slits_left + self.slits_right)
-        # Load archival sky spectrum for absolute correction
-        sky_spectrum, sky_fwhm_pix = flexure.get_archive_spectrum(self.par['flexure']['spectrum'])
+        # # Load archival sky spectrum for absolute correction
```
Comment (rcooke-ast) [inline pypeit/find_objects.py]: These two commented lines can be safely removed  :-)

File: pypeit/core/flexure.py [hunk 2]
```diff
@@ -169,10 +177,27 @@ def spec_flex_shift(obj_skyspec, sky_file, spec_fwhm_pix=None, mxshft=20, excess
 
     msgs.warn("If we use Paranal, cut down on wavelength early on")
 
-    # Load Archival sky spectrum
-    arx_skyspec, arx_fwhm_pix = get_archive_spectrum(sky_file, obj_skyspec=obj_skyspec, spec_fwhm_pix=spec_fwhm_pix)
+    if sky_file is None and arx_skyspec is None:
```
Comment (rcooke-ast) [inline pypeit/core/flexure.py:179]: Perhaps add a comment along the lines of `# Check input mode` or something like that?

File: pypeit/core/flexure.py [hunk 3]
```diff
@@ -169,10 +177,27 @@ def spec_flex_shift(obj_skyspec, sky_file, spec_fwhm_pix=None, mxshft=20, excess
...
+        msgs.error("sky_file or arx_skyspec must be provided")
+    elif sky_file is not None and arx_skyspec is not None:
+        msgs.warn("sky_file and arx_skyspec both provided. Using arx_skyspec.")
+        sky_file = None
+
+    # Arxiv sky spectrum
+    if arx_skyspec is None:
```
Comment (rcooke-ast) [inline pypeit/core/flexure.py]: Would it be slightly clearer is this read:
`if sky_file is not None:`
Then, you could move the code from line 199 below to inside this if statement (i.e. to keep the two "modes" all in one place). Explicitly:
```
if sky_file is not None:
    # Load arxiv sky spectrum
    msgs.info("Loading the arxiv sky spectrum and computing its spectral FWHM")
    arx_skyspec, arx_fwhm_pix = get_archive_spectrum(sky_file, obj_skyspec=obj_skyspec, spec_fwhm_pix=spec_fwhm_pix)
    if sky_file != 'model':
        # get gaussian sigma (pixels) for smoothing
        smooth_fwhm_pix = get_fwhm_gauss_smooth(arx_skyspec, obj_skyspec, arx_fwhm_pix, spec_fwhm_pix=spec_fwhm_pix)

        if smooth_fwhm_pix is None:
            # smooth_fwhm_pix is None if spec_fwhm_pix<0, i.e., the wavelength calibration is bad
            msgs.warn('No flexure correction could be computed for this slit/object')
            return None
        if smooth_fwhm_pix > 0:
            arx_skyspec = arx_skyspec.gauss_smooth(smooth_fwhm_pix)
elif arx_fwhm_pix is not None:
    smooth_fwhm_pix = None
    ...
```
Does this capture the same control flow, but groups the modes together?

**Review by rcooke-ast — APPROVED** [review]
I have some minor comments to consider, but everything looks fixed to me, so I am approving. Thanks very much @debora-pe for turning around this fix so quickly!

File: pypeit/find_objects.py (hunk 1 above)
Comment (debora-pe) [reply pypeit/find_objects.py]: Done

File: pypeit/core/flexure.py (hunk 2 above)
Comment (debora-pe) [reply pypeit/core/flexure.py:179]: Done

File: pypeit/core/flexure.py (hunk 3 above)
Comment (debora-pe) [reply pypeit/core/flexure.py]: Thanks about the suggestion. It actually allowed me to find a small bug! I re-arrange a bit the flow.

File: pypeit/core/flexure.py [hunk 4]
```diff
@@ -169,9 +177,29 @@ def spec_flex_shift(obj_skyspec, sky_file, spec_fwhm_pix=None, mxshft=20, excess
 
     msgs.warn("If we use Paranal, cut down on wavelength early on")
```
Comment (rcooke-ast) [inline pypeit/core/flexure.py]: Sorry - just noticed this... Do you think this should be a comment, rather than a user-facing message?

Comment (rcooke-ast) [reply pypeit/core/flexure.py]: Or perhaps a `# TODO`?

Comment (debora-pe) [reply pypeit/core/flexure.py]: Not really sure what this message is about. It has been here for a long time. I can remove it.

Comment (rcooke-ast) [reply pypeit/core/flexure.py]: That would be fine by me - as you wish!

File: pypeit/core/flexure.py [hunk 5]
```diff
@@ -167,11 +175,32 @@ def spec_flex_shift(obj_skyspec, sky_file, spec_fwhm_pix=None, mxshft=20, excess
 
     # TODO None of these routines should have dependencies on XSpectrum1d!
 
-    msgs.warn("If we use Paranal, cut down on wavelength early on")
-
-    # Load Archival sky spectrum
-    arx_skyspec, arx_fwhm_pix = get_archive_spectrum(sky_file, obj_skyspec=obj_skyspec, spec_fwhm_pix=spec_fwhm_pix)
+    # commenting this msgs out for now. If no objections, we should remove it
+    # msgs.warn("If we use Paranal, cut down on wavelength early on")
```
Comment (kbwestfall) [inline pypeit/core/flexure.py]: Agreed.  I suggest removing it.

**Review by kbwestfall — APPROVED** [review]
Looks good to me!

**Comment by kbwestfall** [comment]
PS - Add something to the change log and update the docs.

File: pypeit/core/flexure.py (hunk 5 above)
Comment (debora-pe) [reply pypeit/core/flexure.py]: removed

**Comment by debora-pe** [comment]
```
========================
Sphinx build successful.
========================
```

**Comment by debora-pe** [comment]
merging

**Pull request merged**